File: frontend/src/components/update-header.js
// Temporary script to update header
const fs = require('fs');
const path = require('path');

const filePath = path.join(__dirname, 'EnhancedHeader.jsx');
let content = fs.readFileSync(filePath, 'utf8');

// Add Wifi to imports if not already there
if (!content.includes('Wifi')) {
  content = content.replace(
    /(Users,\s+Menu,)/,
    'Users,\n  Wifi,\n  Menu,'
  );
}

// Add WebSocket navigation item
const navigationPattern = /(\s+\/\/ Team - Only for managers and admins[\s\S]*?\[\{ id: "team".*?\}\],\s*\))/;
const replacement = `$1
    // WebSocket Streaming - Only for managers and admins
    ...(userRole !== "user" ? [{ id: "websocket", label: "WebSocket Streaming", icon: Wifi }] : []),`;

if (!content.includes('id: "websocket"')) {
  content = content.replace(navigationPattern, replacement);
}

fs.writeFileSync(filePath, content, 'utf8');
console.log('✓ EnhancedHeader.jsx updated with WebSocket navigation');

File: update_header.py
import re

# Read file
with open(r'c:\Users\Admin\Documents\GitHub\BidBuilder\frontend\src\components\EnhancedHeader.jsx', 'r') as f:
    content = f.read()

# 1. Add Wifi to imports
old_imports = """  Users,
  Menu,
  X,
} from "lucide-react" """

new_imports = """  Users,
  Wifi,
  Menu,
  X,
} from "lucide-react" """

content = content.replace(old_imports, new_imports)

# 2. Add WebSocket navigation item
old_nav = """  const navigationItems = [
    // Dashboard - Only for non-user roles
    ...(userRole !== "user" ? [{ id: "dashboard", label: "Dashboard", icon: BarChart3 }] : []),
    { id: "list", label: "Proposals", icon: FileText },
    // Templates - Only for managers and admins
    ...(userRole !== "user" ? [{ id: "templates", label: "Templates", icon: FileTemplate }] : []),
    // Team - Only for managers and admins
    ...(userRole !== "user" ? [{ id: "team", label: "Team", icon: Users }] : []),
  ]"""

new_nav = """  const navigationItems = [
    // Dashboard - Only for non-user roles
    ...(userRole !== "user" ? [{ id: "dashboard", label: "Dashboard", icon: BarChart3 }] : []),
    { id: "list", label: "Proposals", icon: FileText },
    // Templates - Only for managers and admins
    ...(userRole !== "user" ? [{ id: "templates", label: "Templates", icon: FileTemplate }] : []),
    // Team - Only for managers and admins
    ...(userRole !== "user" ? [{ id: "team", label: "Team", icon: Users }] : []),
    // WebSocket Streaming - Only for managers and admins
    ...(userRole !== "user" ? [{ id: "websocket", label: "WebSocket Streaming", icon: Wifi }] : []),
  ]"""

content = content.replace(old_nav, new_nav)

# Write file back
with open(r'c:\Users\Admin\Documents\GitHub\BidBuilder\frontend\src\components\EnhancedHeader.jsx', 'w') as f:
    f.write(content)

print("✓ EnhancedHeader updated with WebSocket support")
